### ISSUE TITLE
Feature: JSON parse options

### ISSUE THREAD

**Issue by Jolanrensen** [Kotlin/dataframe#459]
Currently we just use Klaxon to read JSON using the default settings. We should allow users to change Klaxon settings.

**Comment by zaleslaw** [comment]
From another side, need to do it without expanding any Klaxon API in our public API if it changes in the future

**Comment by Jolanrensen** [comment]
That's probably why it was hidden

**Comment by zaleslaw** [comment]
Some proxies for basic options will be enough, probably?

**Comment by Jolanrensen** [comment]
probably yes. For advanced stuff, users can simply parse it using their own parser, export to string and feed it to DF again

**Comment by dsalv** [comment]
@Jolanrensen is it possible to disable parsing of JSONs and read them as Strings?

**Comment by Jolanrensen** [comment]
Definitely! It depends a bit from where you want to disable this. If you're reading from CSV, for instance, you can supply a `colTypes = mapOf("yourColName" to ColType.String)`, this will skip parsing completely and makes sure a column remains `String`.

The second way to do it is a bit less obvious, but also works well. JSON is parsed to DataRow/DataFrame, meaning, if you add `parserOptions = ParserOptions(skipTypes = setOf(typeOf<AnyFrame>(), typeOf<AnyRow>()))`, the result of parsing will never contain a DataRow or DataFrame so JSON will remain String.

**Comment by dsalv** [comment]
> Definitely! It depends a bit from where you want to disable this. If you're reading from CSV, for instance, you can supply a `colTypes = mapOf("yourColName" to ColType.String)`, this will skip parsing completely and makes sure a column remains `String`.
> 
> The second way to do it is a bit less obvious, but also works well. JSON is parsed to DataRow/DataFrame, meaning, if you add `parserOptions = ParserOptions(skipTypes = setOf(typeOf<AnyFrame>(), typeOf<AnyRow>()))`, the result of parsing will never contain a DataRow or DataFrame so JSON will remain String. This can also be done globally by calling `DataFrame.parser.addSkipType(...)`.

Thank you! Second approach looks interesting.

**Comment by zaleslaw** [comment]
Please check the actuality of this, if yes, propose the plan for implementation. Also, we need this workaround on the site

**Comment by Jolanrensen** [comment]
While we definitely don't use Klaxon anymore (we use Kotlinx Serialization now), we don't expose any parsing options of that.
I would recommend doing something similar to CSV:

- offer a stable API without any Kotlinx Serialization specific features
- have a single overload (like the one which takes a raw `InputStream`) and make it accept a custom `kotlinx.serialization.json.Json {}` instance with extra settings.

We don't implicitly call `.parse()` on our resulting DataFrame when reading JSON (in contract to csv), so adding ParserOptions doesn't make sense, unless we do add that.
